### PR TITLE
Infra - Add script to push new image to Dockerhub

### PR DESCRIPTION
**Pull request by kbendick**
Adds a simple shell script to push the image as a multi-platform image to dockerhub.

This script takes in the tag as an argument, and is set up to make it easy to change to support different repositories etc. It also checks if docker is installed and running, as well as validates the image tag is a valid image tag.

The tag defaults to latest if not provided.

### CODE REVIEW

File: dev/build-and-push-image.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+docker buildx build -t tabulario/spark-iceberg:latest -t tabulario/spark-iceberg:3.2.1_0.13.1 --platform=linux/amd64,linux/arm64 ./spark --push
```
Comment (samredai) [inline dev/build-and-push-image.sh:3]: If we parameterize the tag we could maybe drive this using github tags or branch names, right? Also I think when you exclude the second `-t` it uses the tag name you gave it locally. What do you think of this?:
```
docker buildx build -t tabulario/spark-iceberg:$1 --platform=linux/amd64,linux/arm64 ./spark --push
```

Comment (kbendick) [reply dev/build-and-push-image.sh:3]: Yeah I'm good with that. I mostly threw this together just to have... something to include for our use.

I'd ideally like to check that things are installed and then most importantly check if the argument `$1`  exists. Possibly also that the image tag, outside of `latest` is unique (not sure if that's a goal).

I have some shell scripts that do more or less most of that already, I'll dig them up and then put them on this so we can go about automating the deployments on merge and/or on trigger (with tag specified then).

**Comment by kbendick** [comment]
Output with docker running for default tag:
```bash
➜  ./dev/build-and-push-image.sh
Building and pushing multi-architecture image as tabulario/spark-iceberg:latest for platforms linux/amd64 and linux/arm64
```

Output with docker running for tag overridden:
```bash
➜ ./dev/build-and-push-image.sh -t "latest_and_greatest"
Building and pushing multi-architecture image as tabulario/spark-iceberg:latest_and_greatest for platforms linux/amd64 and linux/arm64
```

Output with invalid tag - first character is invalid:
```bash
➜  ./dev/build-and-push-image.sh -t ".latest" 
The image tag ".latest" is invalid.
  A valid image tag must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes.
  A tag name may not start with a period or a dash and may contain a maximum of 128 characters
```

Output with invalid tag. Second group (all after first character) has invalid characters
```
➜  ./dev/build-and-push-image.sh -t "latest&greatest" 
The image tag "latest&greatest" is invalid.
  A valid image tag must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes.
  A tag name may not start with a period or a dash and may contain a maximum of 128 characters
```

Output for tag being too long (128 characters or more):
```bash
➜ too_long_tag=$(python -c "print('a' * 129)")
➜  echo $too_long_tag
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
➜  ./dev/build-and-push-image.sh -t $too_long_tag      
The image tag "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" is invalid.
  A valid image tag must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes.
  A tag name may not start with a period or a dash and may contain a maximum of 128 characters
```

File: dev/build-and-push-image.sh (hunk 1 above)
Comment (kbendick) [reply dev/build-and-push-image.sh]: I added in the tag argument, with validation.

**Comment by samredai** [comment]
@kbendick is this ready to merge?

**Pull request merged**